cli/script/command-executor.ts

```diff
@@ -1020,6 +1057,50 @@ function getPackageMetricsString(packageObject: PackageWithMetrics): string {
     return returnString;
 }
 
+function getReactNativeProjectAppVersion(platform: string, projectName: string): Promise<string> {
+    if (platform === "ios") {
+        try {
+            var infoPlistContents: string = fs.readFileSync(path.join("iOS", projectName, "Info.plist")).toString();
+        } catch (err) {
+            try {
+                infoPlistContents = fs.readFileSync(path.join("iOS", "Info.plist")).toString();
+            } catch (err) {
+                throw new Error(`Unable to find or read "Info.plist" in the "iOS/${projectName}" or "iOS" folders.`);
+            }
+        }
+        
+        var appVersionRegex: RegExp = /CFBundleShortVersionString\s*<\s*\/key\s*>\s*<\s*string\s*>(.*?)<\s*\/string\s*>/;
+        var parsedInfoPlist: string[] = infoPlistContents.match(appVersionRegex);
+        if (parsedInfoPlist && parsedInfoPlist[1]) {
+             if (semver.valid(parsedInfoPlist[1]) === null) {
+                throw new Error("Please update \"Info.plist\" to use a semver compliant \"CFBundleShortVersionString\", for example \"1.0.3\".");
+            } else {
+                return Q(parsedInfoPlist[1]);
+            }
+        } else {
+            throw new Error("Unable to parse the \"CFBundleShortVersionString\" from \"Info.plist\".");
+        }
+    } else {
+        try {
+            var buildGradleContents: string = fs.readFileSync(path.join("android", "app", "build.gradle")).toString();
+        } catch (err) {
+            throw new Error("Unable to find or read \"build.gradle\" in the \"android/app\" folder.");
+        }
+        
+        var appVersionRegex: RegExp = /versionName\s+"(.*?)"/;
```

Comment:
Sure. In practice, it might not be common, but any number of other plugins or config objects could also have a method call versionName. Additionally, the value being passed to the versionName method could also be dynamic, and therefore, simply capturing the string like this wouldn't be sufficient. Once again, that could be rare, but we already know of one example of a native Android app integrating RN into it, and we should keep that in mind here. The AndroidManifest.xml solution wouldn't have either of these issues, and might merits a quick investigation.